src/Contract/Transaction.purs

```diff
@@ -206,12 +214,54 @@ finalizeTx tx datums redeemers = wrap $ QueryM.finalizeTx tx datums redeemers
 -- | reindex the redeemers with such inputs. This must be crucially called after
 -- | balancing when all inputs are in place so they cannot be reordered.
 reindexSpentScriptRedeemers
-  :: Transaction.Transaction
+  :: Array Transaction.TransactionInput
   -> Array (Transaction.Redeemer /\ Maybe Transaction.TransactionInput)
   -> Contract
        ( Either
            ReindexRedeemersExport.ReindexErrors
            (Array Transaction.Redeemer)
        )
 reindexSpentScriptRedeemers balancedTx =
-  wrap <<< ReindexRedeemers.reindexSpentScriptRedeemers balancedTx
\ No newline at end of file
+  wrap <<< ReindexRedeemers.reindexSpentScriptRedeemers balancedTx
+
+newtype BalancedSignedTransaction = BalancedSignedTransaction
+  { transaction :: Transaction.Transaction -- the balanced and unsigned transaction to help with logging
+  , signedTxCbor :: ByteArray -- the balanced and signed cbor ByteArray representation used in `submit`
+  }
+
+derive instance Generic BalancedSignedTransaction _
+derive instance Newtype BalancedSignedTransaction _
+derive newtype instance Eq BalancedSignedTransaction
+
+instance Show BalancedSignedTransaction where
+  show = genericShow
+
+-- | A helper that wraps a few steps into: balance an unbalanced transaction
+-- | (`balanceTx`), reindex script spend redeemers (not minting redeemers)
+-- | (`reindexSpentScriptRedeemers`), attach datums and redeemers to the
+-- | transaction (`finalizeTx`), and finally sign (`signTransactionBytes`).
+-- | The return type includes the balanced (but unsigned) transaction for
+-- | logging and more importantly, the `ByteArray` to be used with `Submit` to
+-- | submit  the transaction.
+balanceAndSignTx
+  :: UnattachedUnbalancedTx
+  -> Contract (Maybe BalancedSignedTransaction)
+balanceAndSignTx
+  (UnattachedUnbalancedTx { unbalancedTx, datums, redeemersTxIns }) = do
+  -- Balance unbalanced tx:
+  balancedTx <- liftedE' $ balanceTx unbalancedTx
+  log "balanceAndSignTx: Transaction successfully balanced"
```

Comment:
I'm not going to hold up this PR for this, but we're already logging way too much. I'd like to avoid adding any more logging